src/ComputeSharp.D2D1/Shaders/Interop/Factories/Abstract/D2D1TransformMapper{T, TParameters}.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Drawing;
+using System.Runtime.CompilerServices;
+using ComputeSharp.D2D1.Shaders.Interop.Helpers;
+
+namespace ComputeSharp.D2D1.Shaders.Interop.Factories.Abstract;
+
+/// <summary>
+/// A base type for a transform mapper to be used in a D2D1 pixel shader effect.
+/// </summary>
+/// <typeparam name="T">The type of D2D1 pixel shader associated to the transform mapper.</typeparam>
+/// <typeparam name="TParameters">The type of parameters that the transform mapper will use.</typeparam>
+internal abstract class D2D1TransformMapper<T, TParameters> : ID2D1TransformMapper<T>
+    where T : unmanaged, ID2D1PixelShader
+    where TParameters : unmanaged
+{
+    /// <summary>
+    /// The parameters to use in this transform mapper.
+    /// </summary>
+    private TParameters parameters;
+
+    /// <inheritdoc cref="D2D1TransformMapperFactory{T, TSelf, TParameters, TTransformMapper}.Parameters"/>
+    public D2D1TransformMapperParametersAccessor<T, TParameters>? Parameters { get; init; }
```

Comment:
some naming overloading here ... you use the name "parameters" for both the parameters and the parameters accessor. it's kind of confusing. I suggest renaming `Parameters` to e.g. `ParametersAccessor`, or maybe renaming `parameters` to `currentParameters`, or just something that helps to disambiguate between them

Reply:
Yeah good call. Renamed to `ParametersAccessor` in 7dcc2ca22784ad32b32f67125e7f85a7f01b08a6 🙂